Bugfix unable to create user

stlgaits opened the pull request:
Since adding timezone to User Entity, the commands to create a new Myddleware User weren't working anymore, due to missing code to add user timezone to the new object.
This has been fixed in both methods (Add User Command from a terminal + UI Form on /register path).
I've also added code in case there's an error on the UI /register form submit.
This merge also includes an upgrade of some packages, moving on to Symfony 4.4.37.

Myddleware committed:
Init version 3.0.4a

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bugfix : add transaction on method generateDocuments

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bug fix : no search executed if no search parameter set

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Init version 3.0.4b

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bug fix : never send tagerçid for the create function

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Fix bug : checkBeforeDelete didn't work when there was several document without all in error

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
SugarCRM connector : manage deletion
Add formula function : changeMultiValue
DocumentMangaer : add document type when the history is read

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bugfix : fix the URL of the task when the user run an action from the document list view

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
PostgrSQL connecteur : init

Myddleware committed:
Merge branch 'main' into hotfix3

Myddleware committed:
Init version3.0.4c

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Merge branch 'hotfix3' of https://github.com/Myddleware/myddleware.git into hotfix3

Myddleware committed:
Postgres connector : ongoing development

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Database connector : get SQL exceptions

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Deletion document type : change the query to detect if the record can be deleted or not

Signed-off-by: myddleware <stephanefaure@myddleware.com>

CamillePMyddleware committed:
fix button ON/OFF on rule list

CamillePMyddleware committed:
Fix SQL error, when you click on the + sign in the rule creation

CamillePMyddleware committed:
visual bug: when double clicking a field on the field mapping, the placeholder is replaced with a grey background and the arrow is loading indefinetely on hover

CamillePMyddleware committed:
Fix Error chart : number not diplayed properly when > 1000 - #586

Myddleware committed:
Bug fix : Deleted document not filtered for create only rule

Signed-off-by: myddleware <stephanefaure@myddleware.com>

CamillePMyddleware committed:
Display edit for: Display error, the dates overlap #602

CamillePMyddleware committed:
Merge branch 'hotfix3' of https://github.com/Myddleware/myddleware.git into hotfix3

Myddleware committed:
Formula add parameter to return null if no value found otherwise, the formula returns an exception

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Merge branch 'hotfix3' of https://github.com/Myddleware/myddleware.git into hotfix3

Myddleware committed:
Rerun action : fix issue when a status is redefined

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Massaction don't stop anymore when en arror happens

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bug fix : A new record is created when a data deleted by Myddleware is restored into the source application

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Add default php value in config table

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Upgrade : manage composer install command if php isn't the default php version

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Merge branch 'Release-3.0.5a' into hotfix3

Myddleware committed:
Bugfix : Return an error if reference date doesn't change after a read action

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Force composer to run with php 7.4 if several php version are installed on the server

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Change the relate search in case of a document has been previously deleted

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Change postgresql logo
Allow "." in the field name

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bug fix : get the history from the current process instead of getting it from the database

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Enhancement : improve error message when transform failed

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bug fix : rule params weren't refreshed when several are executed using the command synchro ALL

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bug fix : change the order parameter to get the relate document

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
No relate check for deletion document. The document linked could be also deleted.

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Enhancement : allows some customizations on the default controller

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Enhancement : allows tools and formulaFunction classes to be redefined

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Bugfix : function declared 2 times

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Enhancement to be able to redefine more action (create specific formula for example)
Bug fix : Fix the table name with capital letter
Database connector : don't add deletion field in case of history reading

Signed-off-by: myddleware <stephanefaure@myddleware.com>

stlgaits committed:
add UTC timezone to User entity when using myddleware:add-user command

stlgaits committed:
add UTC default timezone to user when registering from app_register (/install UI)

stlgaits committed:
default UTC timezone on User Entity + bugfix image not found PostgreSQL.png

stlgaits committed:
add .php-version file to force use of correct PHP version (list of all version available on your computer can be checked with symfony local:php:list)

stlgaits committed:
change webpack config (build path) & update install requirements info

Myddleware commented on webpack.config.js:
```diff
@@ -10,7 +10,7 @@ Encore
     .setOutputPath('public/build/')
     // public path used by the web server to access the output path
-    .setPublicPath('./build')
+    .setPublicPath('/build')
```
@EstelleMyddleware , are you sure it is safe to push this change to main branch ?

Myddleware reviewed (commented):
Please @EstelleMyddleware  could you also change the Myddleware version here, should be 3.0.5b : https://github.com/Myddleware/myddleware/blob/6e2da2b61dfe1872358381183e679c935dcb749e/.env#L1

Thanks.

Myddleware commented on composer.json:
```diff
@@ -14,7 +14,7 @@
     "require": {
         "php": ">=7.4",
-        "symfony/symfony": "^4.4.30",
+        "symfony/symfony": "^4.4.37",
```
@EstelleMyddleware , we should avoid to upgrade Symfony on a hotfix version if it is possible. 
If we have to upgrade Symfony, we should request non regression tests from @RomanFaure .

stlgaits replied:
@Myddleware Normally it should be safe to upgrade as it is only a minor version https://symfony.com/doc/current/contributing/community/releases.html#backward-compatibility

Myddleware committed:
Prestashop connector : Fix the list option for Prestashop 's fields

Signed-off-by: myddleware <stephanefaure@myddleware.com>

Myddleware committed:
Update .env

Init version 3.0.6a

Myddleware committed:
Add the possibility to change the user's timezone (+ automatically change the times in the documents and tasks according to the user's timezone (#625)

Co-authored-by: cpichaud <camillepichaud@myddleware.com>

Myddleware committed:
Rule duplicate action (#626)

* Hotfix3 (#605)

* Bugfix : 
Add transaction on method generateDocuments
No search executed if no search parameter set
Never send tagerçid for the create function
CheckBeforeDelete didn't work when there was several document without all in error
Fix the URL of the task when the user run an action from the document list view
Fix button ON/OFF on rule list
Fix SQL error, when you click on the + sign in the rule creation
Visual bug: when double clicking a field on the field mapping, the placeholder is replaced with a grey background and the arrow is loading indefinetely on hover
Fix Error chart : number not diplayed properly when > 1000 - #586
Deleted document not filtered for create only rule
Display edit for: Display error, the dates overlap #602
Rerun action : Fix issue when a status is redefined
A new record is created when a data deleted by Myddleware is restored into the source application

* Enhancement : 
Add formula function : changeMultiValue
DocumentMangaer : add document type when the history is read
Deletion document type : change the query to detect if the record can be deleted or not
Formula changeValue function : add parameter to return null if no value found otherwise, the formula returns an exception
Massaction don't stop anymore when en arror happens
Upgrade : manage composer install command if php isn't the default php version

* Connector  : 
* SugarCRM : manage deletion
* PostgrSQL new connector
* Database : get SQL exceptions

* Upgrade symfony 4.4.18 (#606)

* resize container for big screens on install requirements page

* adjust menu position, especially for larger screens

* bugfix: attempt to setup docker configuration to operational, not working yet, still some issues with nginx and php-fpm

* update vendors & stop tracking unnecessary files

* remove unused files

* git ignore old files

* upgrade node_modules

* change Myddleware version

* update vendors & Myddleware version 3.0.2

* Fix bug on data history before sending data

* Add the possibility to change the user's timezone (+ automatically change the times in the documents and tasks according to the user's timezone (#607)

Co-authored-by: cpichaud <camillepichaud@myddleware.com>

* Change version to 3.0.5a

* Remove a declaration of changeMultiValue function

* Rollback upgrade command on composer install

* Added a duplicate rule function and a duplicate rule button in the information of a rule

* Add select behavior in the relationship tab when creating a rule

* Minor modification on duplicate rule button, add try/catch

Co-authored-by: cpichaud <camillepichaud@myddleware.com>

Myddleware committed:
Merge branch 'Release-3.0.6' into composer-upgrade

stlgaits committed:
remove facebook graph sdk from required packages https://github.com/Myddleware/myddleware/issues/618#issuecomment-1027289547
